lambdas/process-journey-event/src/main/java/uk/gov/di/ipv/core/processjourneyevent/statemachine/StateMachine.java

```diff
@@ -27,11 +31,27 @@ public State transition(String startState, String event, JourneyContext journeyC
                     String.format("Unknown state provided to state machine: %s", startState));
         }
 
+        if (currentPage != null && state instanceof BasicState basicState) {
+            if (isPageOrCriStateAndOutOfSync(basicState, currentPage)) {
+                return state;
+            } else if (basicState.getResponse() instanceof ProcessStepResponse) {
+                throw new UnknownStateException(
+                        String.format("Unknown state provided to state machine: %s", startState));
```

Comment:
I think this error message isn't quite right - should it be something like `Unexpected page event received in process state`?